sklearn/cluster/optics_.py

```diff
@@ -222,7 +223,9 @@ def fit(self, X, y=None):
         Parameters
         ----------
         X : array, shape (n_samples, n_features), or (n_samples, n_samples)  \
-if metric=’precomputed’.
+if metric=’precomputed’, or sparse matrix (n_samples, n_features) if metric
+            in ['cityblock', 'cosine', 'euclidean', 'haversine', 'l2', 'l1',
```

Comment:
Please keep the first line of the `X`'s description short and as usual syntax, and put the information in the comments of the parameter.